corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,17 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects
+        .accessible_to_user(domain, user)
+        .values_list('id', flat=True))
+
+    return other_user.sql_locations.filter(id__in=accessible_location_ids).count() > 0
```

Comment:
ok, will do. i built most of this with the assumption that these queries will only work against commcare users. what is the status of webusers and locations? should that be first class support?